bindings/node/lib/clientEncryption.js

```diff
@@ -202,12 +165,39 @@ module.exports = function(modules) {
      * });
      */
     createDataKey(provider, options, callback) {
-      if (typeof options === 'function') (callback = options), (options = {});
+      if (typeof options === 'function') {
```

Comment:
I think the majority of the changes in this file are whitespace/style changes, and it made it really confusing to review. It looks like you could make a one-line change on the original L210 to:
```
const context = this._mongoCrypt.makeDataKeyContext(dataKeyBson, options);
```
and with your C++ changes the `keyAltNames` would be applied. Is that correct?

Reply:
Indeed the changes could've been less than what's here. I standby the decision to make things easier to read/debug for the next engineer. Is there something specific that you think is less optimal here?

Reply:
I won't comment on whether these changes subjectively improve readability/debugability for future developers, but I can say for sure that changing it this way makes it harder to review and easier to introduce new bugs.

I anchored this comment to splitting the one-liner into a few lines, but I was more concerned about `sanitizeDataKeyOptions`. It _seems_ like you copied the contents into this function with some subtle changes (for-loop => map I think was one change). I read it a few times, but I couldn't tell you for sure whether there's a new bug there or not, or that the bug that I see in the old code is gone for sure.

If you're confident that it's a net improvement in code quality that's fine, but I'd ask that you consider the implications in future changes

Reply:
Thanks for the insight, I do agree that when addressing a bug fix it would be ideal to have just what is necessary to represent the fix in isolation from any refactor. Unfortunately this fix also required orchestrating new CI tasks/scripts to prove the code changes were, in fact, effective which required some debugging. Making the sanitization more functional was helpful as opposed to the key deletion and new array creating that then overwrote the old one logic.  
[We do have tests covering this usage here which tests for the proper errors as well](https://evergreen.mongodb.com/task/mongo_node_driver_ubuntu1804_custom_csfle_tests_run_custom_csfle_tests_51e976861ebc4a0e8b8501719656effeb7ed522e_21_06_22_14_05_14).